internal/builders/go/pkg/build_test.go

```diff
@@ -1047,3 +1048,114 @@ func Test_generateCommand(t *testing.T) {
 func asPointer(s string) *string {
 	return &s
 }
+
+func TestGoBuild_Run(t *testing.T) {
+	type fields struct {
+		cfg     *GoReleaserConfig
+		goc     string
+		argEnv  map[string]string
+		ldflags string
+	}
+	type args struct {
+		dry bool
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		wantErr bool
+	}{
+		{
+			name: "valid flags",
+			fields: fields{
+				cfg: &GoReleaserConfig{
+					Goos:   "linux",
+					Goarch: "amd64",
+					Binary: "binary",
+					Main:   asPointer("../builders/go/main.go"),
+					Dir:    asPointer("../builders/go"),
+					Ldflags: []string{
+						"-X main.version=1.0.0",
+					},
+				},
+			},
+			args: args{
+				dry: true,
+			},
+		},
+		{
+			name: "valid flags",
+			fields: fields{
+				cfg: &GoReleaserConfig{
+					Goos:   "linux",
+					Goarch: "amd64",
+					Binary: "/tmp/binary",
+					Main:   asPointer("main.go"),
+					Dir:    asPointer("./testdata/go"),
+					Ldflags: []string{
+						"-X main.version=1.0.0",
+					},
+				},
+			},
+			args: args{
+				dry: false,
+			},
+		},
+		{
+			name: "slash in the binary name",
+			fields: fields{
+				cfg: &GoReleaserConfig{
+					Goos:   "linux",
+					Goarch: "amd64",
+					Binary: "tmp/binary",
+					Main:   asPointer("../builders/go/main.go"),
+					Dir:    asPointer("../builders/go"),
+				},
+			},
+			args: args{
+				dry: true,
+			},
+			wantErr: true,
+		},
+		{
+			name: "invalid flags",
+			fields: fields{
+				cfg: &GoReleaserConfig{
+					Goos:    "linux",
+					Goarch:  "amd64",
+					Binary:  "binary",
+					Main:    asPointer("../builders/go/main.go"),
+					Dir:     asPointer("../builders/go"),
+					Ldflags: []string{},
+				},
+			},
+			args: args{
+				dry: true,
+			},
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			b := &GoBuild{
+				cfg:     tt.fields.cfg,
+				goc:     tt.fields.goc,
+				argEnv:  tt.fields.argEnv,
+				ldflags: tt.fields.ldflags,
+			}
+			defer os.Unsetenv("OUT_BINARY")
+			os.Setenv("OUTPUT_BINARY", tt.fields.cfg.Binary)
```

Comment:
Use t.Setenv so that the env var is cleaned up automatically.

Reply:
Agreed. Thanks